### ISSUE TITLE
Window can't be focused when opened from menubar icon

### ISSUE THREAD

**Issue by ajhall** [kontena/lens#6682]
**Describe the bug**
When opening Lens from the menubar icon on macOS, the window is in a weird state where the OS doesn't treat it as the "active" app. For example, the menubar doesn't change to show the Lens app name or menus and the dock icon doesn't display a dot to show that the app is running.

**To Reproduce**
1. Run Lens from the dock icon. The main window opens, the running indicator (black dot) appears, and the menubar icon appears.
2. Look at the menubar. The app name shows Lens and the menus are for Lens.
3. Close the main Lens window by clicking the red X or pressing cmd+shift+w. The window closes and black dot disappears, but the menubar icon persists.
4. Open the main Lens window again by clicking the menubar icon > Open Lens, or by clicking the dock icon again.
5. Focus the main Lens window and look at the menubar. The app name and menus are for whatever app you were previously using, and not for Lens. The running indicator is also missing from the dock.

**Expected behavior**
When Lens is opened from the menubar app, it should be possible to use its menus from the macOS menu bar. It should also show the running indicator under its icon in the dock.

**Screenshots**
<img width="932" alt="image" src="https://user-images.githubusercontent.com/1380292/204873601-c16b7d46-0c46-40ad-a3cf-f3923be69b49.png">

**Environment (please complete the following information):**
- Lens Version: OpenLens 6.2.0 or later
- OS: macOS Ventura 13.0.1
- Installation method (e.g. snap or AppImage in Linux): MacPorts (built from source)

**Logs:**
These logs show me opening the app, closing the window, opening a new window, and closing the app.
```
❯ /Applications/MacPorts/OpenLens.app/Contents/MacOS/OpenLens
(node:96891) [DEP0111] DeprecationWarning: Access to process.binding('http_parser') is deprecated.
(Use `OpenLens --trace-deprecation ...` to show where the warning was created)
info:    ▪ [CLUSTER-STORE]: LOADING from  ... +0ms
objc[96892]: Class WebSwapCGLLayer is implemented in both /System/Library/Frameworks/WebKit.framework/Versions/A/Frameworks/WebCore.framework/Versions/A/Frameworks/libANGLE-shared.dylib (0x24197f2e0) and /Applications/MacPorts/OpenLens.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Libraries/libGLESv2.dylib (0x10b071348). One of the two will be used. Which one is undefined.
info:    ▪ [CLUSTER-STORE]: LOADED from /Users/andy.hall/Library/Application Support/OpenLens/lens-cluster-store.json +684ms
info:    ▪ 🔌 Starting LensProxy +5ms
info:    ▪ [LENS-PROXY]: Proxy server has started at 127.0.0.1:53627 +2ms
info:    ▪ 🔎 Testing LensProxy connection ... +0ms
info:    ▪ [USER-STORE]: LOADING from  ... +19ms
info:    ▪ [USER-STORE]: LOADED from /Users/andy.hall/Library/Application Support/OpenLens/lens-user-store.json +9ms
info:    ▪ ⚡ LensProxy connection OK +15ms
info:    ▪ [CREATE-ELECTRON-WINDOW]: Loading content for window "splash" from file: /Applications/MacPorts/OpenLens.app/Contents/Resources/static/splash.html... +64ms
info:    ▪ [CREATE-ELECTRON-WINDOW]: Window "splash" loaded +82ms
info:    ▪ [EXTENSIONS-STORE]: LOADING from  ... +5ms
info:    ▪ [EXTENSIONS-STORE]: LOADED from /Users/andy.hall/Library/Application Support/OpenLens/lens-extensions.json +1ms
info:    ▪ [FILESYSTEM-PROVISIONER-STORE]: LOADING from  ... +0ms
info:    ▪ [FILESYSTEM-PROVISIONER-STORE]: LOADED from /Users/andy.hall/Library/Application Support/OpenLens/lens-filesystem-provisioner-store.json +2ms
info:    ▪ [WEBLINK-STORE]: LOADING from  ... +2ms
info:    ▪ [WEBLINK-STORE]: LOADED from /Users/andy.hall/Library/Application Support/OpenLens/lens-weblink-store.json +2ms
info:    ▪ 📟 Setting protocol client for lens:// +3ms
info:    ▪ 📟 Protocol client register succeeded ✅ +1ms
info:    ▪ 🧩 Initializing extensions +0ms
info:    ▪ 🐚 Syncing shell environment +2ms
info:    ┏ [UNIX-SHELL-ENV]: running against /opt/local/bin/fish +1ms
info:    ┃ [1] {
info:    ┃ [2]   command: '',
info:    ┃ [3]   shellArgs: [
info:    ┃ [4]     '-l',
info:    ┃ [5]     '-c',
info:    ┃ [6]     `'/Applications/MacPorts/OpenLens.app/Contents/MacOS/OpenLens' -p '"5f08650dca6949608769d4c5fdfbaa47" + JSON.stringify(process.env) + "5f08650dca6949608769d4c5fdfbaa47"'`
info:    ┃ [7]   ]
info:    ┗ [8] }
info:    ▪ [HOTBAR-STORE]: LOADING from  ... +16ms
info:    ▪ [HOTBAR-STORE]: LOADED from /Users/andy.hall/Library/Application Support/OpenLens/lens-hotbar-store.json +1ms
info:    ▪ [EXTENSION-DISCOVERY] loading extensions from /Users/andy.hall/Library/Application Support/OpenLens +3ms
info:    ┏ [UNIX-SHELL-ENV]: got the following output +163ms
info:    ┃ [1] {
info:    ┃ [2]   rawOutput: '5f08650dca6949608769d4c5fdfbaa47{"PWD":"/Users/andy.hall/Repos/infra-clusters","ELECTRON_RUN_AS_NODE":"1","ELECTRON_NO_ATTACH_CONSOLE":"1","TFENV_CONFIG_DIR":"/Users/andy.hall/.tfenv","HOMEBREW_NO_AUTO_UPDATE":"1","TERM_PROGRAM_VERSION":"3.5.0beta7","TERM_PROGRAM":"iTerm.app","FLUX_SYSTEM_NAMESPACE":"flux","HOMEBREW_CASK_OPTS":"--no-quarantine","SSH_AUTH_SOCK":"/Users/andy.hall/.gnupg/S.gpg-agent.ssh","TFENV_ARCH":"arm64","fish_complete_path":"/opt/homebrew/share/fish/vendor_completions.d /opt/homebrew/share/fish/vendor_completions.d /Users/andy.hall/.config/fish/completions /opt/local/etc/fish/completions /Users/andy.hall/.local/share/fish/vendor_completions.d /opt/local/share/fish/vendor_completions.d /opt/local/share/fish/completions /Users/andy.hall/.local/share/fish/generated_completions /opt/local/share/fish/completions","COLORFGBG":"0;7","HOMEBREW_NO_ENV_HINTS":"1","PATH":"/Users/andy.hall/bin:/Users/andy.hall/.config/fish/conf.d/bin:/Users/andy.hall/.krew/bin:/Users/andy.hall/.config/fish/conf.d/bin:/Users/andy.hall/Library/Python/3.10/bin:/Users/andy.hall/.config/fish/conf.d/bin:/opt/local/libexec/gnubin:/Users/andy.hall/.config/fish/conf.d/bin:/opt/local/sbin:/Users/andy.hall/.config/fish/conf.d/bin:/opt/local/bin:/Users/andy.hall/.config/fish/conf.d/bin:/Users/andy.hall/.azure/bin:/Users/andy.hall/.config/fish/conf.d/bin:/Users/andy.hall/Library/Python/3.9/bin:/Users/andy.hall/.config/fish/conf.d/bin:/Users/andy.hall/.local/bin:/Users/andy.hall/.config/fish/conf.d/bin:/opt/homebrew/sbin:/Users/andy.hall/.config/fish/conf.d/bin:/Users/andy.hall/go/bin:/Users/andy.hall/.config/fish/conf.d/bin:/opt/homebrew/bin:/Users/andy.hall/.config/fish/conf.d/bin:/usr/local/bin:/Users/andy.hall/.config/fish/conf.d/bin:/System/Cryptexes/App/usr/bin:/Users/andy.hall/.config/fish/conf.d/bin:/usr/bin:/Users/andy.hall/.config/fish/conf.d/bin:/bin:/Users/andy.hall/.config/fish/conf.d/bin:/usr/sbin:/Users/andy.hall/.config/fish/conf.d/bin:/sbin:/Users/andy.hall/.config/fish/conf.d/bin:/Library/Apple/usr/bin:/Users/andy.hall/.config/fish/conf.d/bin:/Applications/VMware Fusion Tech Preview.app/Contents/Public:/Users/andy.hall/.config/fish/conf.d/bin:/usr/local/MacGPG2/bin:/Users/andy.hall/.config/fish/conf.d/bin:/Users/andy.hall/.asdf/shims:/Users/andy.hall/.config/fish/conf.d/bin:/opt/local/share/asdf/bin:/Users/andy.hall/.config/fish/conf.d/bin:/Users/andy.hall/.config/fish/conf.d/bin:/Users/andy.hall/.config/fish/conf.d/bin","USE_GKE_GCLOUD_AUTH_PLUGIN":"True","ITERM_SESSION_ID":"w0t0p0:B4AB6E4C-B67D-4594-9AA7-2F878057436D","ORIGINAL_XDG_CURRENT_DESKTOP":"undefined","COMMAND_MODE":"unix2003","ASDF_DIR":"/opt/local/share/asdf","LOGNAME":"andy.hall","TMPDIR":"/var/folders/nm/cbhsq3jj0v51d12nvydv2f400000gq/T/","TERM":"screen-256color-bce","VISUAL":"code --wait","ITERM_PROFILE":"Andy","LC_TERMINAL_VERSION":"3.5.0beta7","XPC_SERVICE_NAME":"0","__CF_USER_TEXT_ENCODING":"0x1F7:0x0:0x0","COLORTERM":"truecolor","EDITOR":"code --wait","TERMINFO_DIRS":"/Applications/iTerm.app/Contents/Resources/terminfo:/usr/share/terminfo","LANG":"en_US.UTF-8","XPC_FLAGS":"0x0","USER":"andy.hall","LC_TERMINAL":"iTerm2","STARSHIP_SESSION_KEY":"6222221884971113","TERM_SESSION_ID":"w0t0p0:B4AB6E4C-B67D-4594-9AA7-2F878057436D","SHELL":"/opt/local/bin/fish","__CFBundleIdentifier":"com.googlecode.iterm2","GOPATH":"/Users/andy.hall/go","GPG_TTY":"not a tty","STARSHIP_SHELL":"fish","SHLVL":"1","HOME":"/Users/andy.hall","KUBECTX_IGNORE_FZF":"1","HOMEBREW_NO_INSTALL_CLEANUP":"1"}5f08650dca6949608769d4c5fdfbaa47\n'
info:    ┗ [3] }
info:    ▪ [SHELL-SYNC]: Synced shell env +1ms
info:    ▪ [EXTENSION-INSTALLER] installing dependencies at /Users/andy.hall/Library/Application Support/OpenLens +5ms
info:    ▪ [EXTENSION-INSTALLER] dependencies installed at /Users/andy.hall/Library/Application Support/OpenLens +342ms
info:    ▪ [EXTENSION-DISCOVERY] watching extension add/remove in /Users/andy.hall/.k8slens/extensions +0ms
info:    ▪ [CREATE-ELECTRON-WINDOW]: Loading content for window "first-application-window" from url: http://localhost:53627... +15ms
info:    ▪ [CREATE-ELECTRON-WINDOW]: Window "first-application-window" loaded +864ms
info:    ▪ [KUBECONFIG-SYNC]: starting requested syncs +155ms
info:    ┏ [KUBECONFIG-SYNC]: starting sync of file/folder +0ms
info:    ┗ [1] { filePath: '/Users/andy.hall/Library/Application Support/OpenLens/kubeconfigs' }
info:    ┏ [KUBECONFIG-SYNC]: starting sync of file/folder +0ms
info:    ┗ [1] { filePath: '/Users/andy.hall/.kube' }
warn:    ┏ [KUBECONFIG-SYNC]: encountered errors while loading config: "value" must be of type object +13ms
warn:    ┃ [ 1] {
warn:    ┃ [ 2]   filePath: '/Users/andy.hall/.kube/kubectx',
warn:    ┃ [ 3]   details: [
warn:    ┃ [ 4]     {
warn:    ┃ [ 5]       message: '"value" must be of type object',
warn:    ┃ [ 6]       path: [],
warn:    ┃ [ 7]       type: 'object.base',
warn:    ┃ [ 8]       context: { type: 'object', label: 'value', value: 'prod' }
warn:    ┃ [ 9]     }
warn:    ┃ [10]   ]
warn:    ┗ [11] }
info:    ▪ [CREATE-ELECTRON-WINDOW]: Loading content for window "splash" from file: /Applications/MacPorts/OpenLens.app/Contents/Resources/static/splash.html... +9s
info:    ▪ [CREATE-ELECTRON-WINDOW]: Window "splash" loaded +90ms
info:    ▪ [CREATE-ELECTRON-WINDOW]: Loading content for window "first-application-window" from url: http://localhost:53627... +7ms
info:    ▪ [CREATE-ELECTRON-WINDOW]: Window "first-application-window" loaded +826ms
info:    ▪ SERVICE:QUIT +7s
```

**Kubeconfig:**
I don't think this is relevant here, but I can provide a sanitized kubeconfig upon request.

**Additional context**
I'm fairly sure this started with version 6.2.0.

**Comment by ajhall** [comment]
This seems to be fixed in 6.3.0, at least on my machine. I'll leave this open in case one of the maintainers would like to confirm.

**Comment by ajhall** [comment]
Seems like https://github.com/lensapp/lens/issues/6756 is a duplicate of this and it was fixed in https://github.com/lensapp/lens/pull/6770. Closing this issue.
